src/lib/y2storage/proposal/autoinst_nfs_planner.rb

```diff
@@ -0,0 +1,212 @@
+      #
+      # Errors are registered when values are missing.
+      #
+      # @param partition_section [AutoinstProfile::PartitionSection]
+      # @param format [:new, :old] whether the section is using the new or old AutoYaST style
+      #
+      # @return [Boolean]
+      def missing_partition_values?(partition_section, format)
+        missing_any_value?(partition_section, mandatory_partition_values(format))
+      end
+
+      # Whether any of the given values is missing in the given section
+      #
+      # Note: finding the first missing value is faster, but all values are checked to
+      # register all possible issues.
+      #
+      # @param section [AutoinstProfile::SectionWithAttributes]
+      # @param values [Array<Symbol>]
+      #
+      # @return [Boolean]
+      def missing_any_value?(section, values)
+        values.map { |v| missing_value?(section, v) }.any?
+      end
+
+      # Whether the given value is missing in the given section
+      #
+      # An error is registered when the value is missing.
+      #
+      # @param section [AutoinstProfile::SectionWithAttributes]
+      # @param value [Symbol]
+      #
+      # @return [Boolean]
+      def missing_value?(section, value)
+        return false if section.send(value)
+
+        issues_list.add(:missing_value, section, value)
+
+        true
+      end
+
+      # Mandatory values for the drive section
+      #
+      # @param format [:new, :old] whether the section is using the new or old AutoYaST style
+      # @return [Array<Symbol>]
+      def mandatory_drive_values(format)
+        mandatory_values(format)[:drive]
+      end
+
+      # Mandatory values for the partition section
+      #
+      # @param format [:new, :old] whether the section is using the new or old AutoYaST style
+      # @return [Array<Symbol>]
+      def mandatory_partition_values(format)
+        mandatory_values(format)[:partition]
+      end
+
+      # Mandatory values depending on the AutoYaST style
+      #
+      # @param format [:new, :old] new or old AutoYaST style
+      # @return [Hash<Symbol, Array<Symbol>>]
+      def mandatory_values(format)
+        if format == :new
+          NEW_FORMAT_MANDATORY_VALUES
+        else
+          OLD_FORMAT_MANDATORY_VALUES
+        end
+      end
+
+      # Name of the server from a NFS share
+      #
+      # @param share [String] e.g., "192.168.56.1:/root_fs"
+      # @return [String]
+      def server(share)
+        server_and_path(share).first || ""
+      end
+
+      # Name of the shared directory from a NFS share
+      #
+      # @param share [String] e.g., "192.168.56.1:/root_fs"
+      # @return [String]
+      def path(share)
+        server_and_path(share).last || ""
+      end
+
+      # Name of the server and the shared directory from a NFS share
+      #
+      # @param share [String] e.g., "192.168.56.1:/root_fs"
+      # @return [Array<String>]
+      def server_and_path(share)
+        share.split(":")
```

Comment:
can happen that there is multiple collong? and is it correct in that case? what about  nfs with password like `jreidinger:ubercoolpassword@192.168.56.1:/root_fs` can value look like this? Maybe is possible to simple prepend `nfs://` and use url parsing there? or can you link to documentation how nfs share can look like?

Reply:
ah, username and password looks like not possible. Please link to `https://tools.ietf.org/html/rfc2224` so we know in future how this url can look like.

Reply:
ah, but I found that path can be omitted, so I worry in that case it won't work.

Reply:
Ok, fixed.